<?xml version="1.0"?>
<launch>

<?ignore  Environment ?>
<arg name="environment" default="gazebo"/>



<?ignore  Trajectory parameters ?>
<arg name="trajectory_type" default="static"/>
<arg name="v_max" default="12.0"/>
<arg name="a_max" default="12.0"/>
<arg name="training" default="0"/>
<arg name="training_trajectories_count" default="0"/>

<?ignore  GP parameters ?>
<arg name="use_gp" default="0"/>
<arg name="gp_from_file" default="0"/>
<arg name="gp_path" default="../outputs/gazebo_simulation/gp_models"/>
<arg name="n_basis_vectors" default="20"/>

<?ignore  MPC parameters ?>
<arg name="t_lookahead" default="1.0"/>
<arg name="n_nodes" default="5"/>

<?ignore  etc ?>
<arg name="explore" default="0"/>
<arg name="log_filename_base" default="trajectory"/>

  <group ns="mpcros">
      <node name="mpc_controller" pkg="mpcros" type="mpc_controller_node.py">
        <param name="environment" value="$(arg environment)"/>

        <param name="trajectory_type" value="$(arg trajectory_type)"/>
        <param name="v_max" value="$(arg v_max)"/>
        <param name="a_max" value="$(arg a_max)"/>
        <param name="use_gp" value="$(arg use_gp)"/>
        <param name="gp_from_file" value="$(arg gp_from_file)"/>
        <param name="gp_path" value="$(arg gp_path)"/>
        <param name="n_basis_vectors" value="$(arg n_basis_vectors)"/>
        <param name="log_filename_base" value="$(arg log_filename_base)"/>
        <param name="training" value="$(arg training)"/>

        <param name="t_lookahead" value="$(arg t_lookahead)"/>
        <param name="n_nodes" value="$(arg n_nodes)"/>

        <param name="explore" value="$(arg explore)"/>
        <param name="training_trajectories_count" value="$(arg training_trajectories_count)"/>
      </node>
      <node name="trajectory_generator" pkg="mpcros" type="trajectory_generator_node.py">
        
      </node>
  </group>

</launch>

